File: since2020/ZeroJudge/ZeroJudge b430.py
'''
	@judge ZeroJudge
	@id b430
	@name 簡單乘法

	@tag Modulus
'''
from sys import stdin

print('\n'.join([ (lambda a, b, n: str(a * b % n))(*map(int, line.split())) for line in stdin ]))

File: since2020/ZeroJudge/ZeroJudge b430.cpp
/**
 * @judge ZeroJudge
 * @id b430
 * @name 簡單乘法
 * 
 * @tag Modulus, Exponentiation by Squaring
 */
#include <bits/stdc++.h>
using namespace std;

uint64_t a, b, n;

uint64_t solve() {
	uint64_t res;
	res = 0;
	if (a >= n) a %= n;
	if (b >= n) b %= n;
	for (res = 0; a; a >>= 1) {
		if (a & 1)
			res = (res + b) % n;
		b = (b + b) % n;
	}
	return res;
}

int main() {
	while (~scanf("%ju%ju%ju", &a, &b, &n))
		printf("%ju\n", solve());
}